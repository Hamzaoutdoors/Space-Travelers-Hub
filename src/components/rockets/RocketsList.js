import React, { useEffect } from 'react';
import { Container } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import { getRockets } from '../../redux/rockets/rockets';
import Rocket from './Rocket';

const Rockets = () => {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rockets);

  useEffect(() => {
    dispatch(getRockets());
  }, [dispatch]);

  const rocketsList = rockets.map((rocket) => {
    const {
      id, name, description, flickrImage,
    } = rocket;
    return (
      <Rocket
        name={name}
        description={description}
        flickrImage={flickrImage}
        key={id}
      />
    );
  });
  return (
    <>
      <Container>
        {rocketsList}
      </Container>
    </>
  );
};

export default Rockets;
