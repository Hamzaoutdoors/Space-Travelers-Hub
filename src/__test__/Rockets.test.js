import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import RocketsList from '../components/rockets/RocketsList';
import store from '../redux/configureStore';

test('rockets list renders properly', () => {
  const tree = renderer
    .create(<Provider store={store}><RocketsList /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
