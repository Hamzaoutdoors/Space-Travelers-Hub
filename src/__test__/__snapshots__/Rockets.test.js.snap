// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rockets list renders properly 1`] = `
<div
  className="container"
>
  <div
    style={
      Object {
        "opacity": 0,
      }
    }
  />
</div>
`;
